File: src/main/java/cz/cvut/kbss/spipes/util/JopaPersistenceUtils.java
package cz.cvut.kbss.spipes.util;

import cz.cvut.kbss.jopa.model.EntityManager;
import org.openrdf.repository.Repository;

/**
 * Created by Miroslav Blasko on 4.1.17.
 */
public class JopaPersistenceUtils {

    public static Repository getRepository(EntityManager entityManager) {
        try {
            return entityManager.unwrap(Repository.class);
        } finally {
            entityManager.close();
        }
    }
}

File: src/main/scala/cz/cvut/kbss/spipes/util/JopaPersistenceUtils.scala
package cz.cvut.kbss.spipes.util

import cz.cvut.kbss.jopa.model.EntityManager
import org.apache.jena.query.Dataset

/**
  * Created by Yan Doroshenko (yandoroshenko@protonmail.com) on 02.08.2017.
  */
object JopaPersistenceUtils {
  def getDataset(entityManager: EntityManager): Dataset =
    try
      entityManager.unwrap(classOf[Dataset])
    finally
      entityManager.close()
}